mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -427,7 +436,7 @@ func resourceMongoDBAtlasClusterCreate(d *schema.ResourceData, meta interface{})
 
 	// We need to validate the oplog_size_mb attr of the advanced configuration option to show the error
 	// before that the cluster is created
-	if oplogSizeMB, ok := d.GetOk("advanced_configuration.oplog_size_mb"); ok {
+	if oplogSizeMB, ok := d.GetOk("advanced_configuration.0.oplog_size_mb"); ok {
```

Comment:
Is this list always guaranteed to be length 1? Want to make sure we don't get any null pointers or index out of bounds.

Reply:
@gmlp can you comment on this?

Reply:
we have the Max Items =1 with this we make sure than more that maximum 1 element can be passed. https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/221/files#diff-104ff45c049e7dbfb15899b74adb243dL292-L293 and with GetOk method, we ensure that the element 0 exists. I can tweak this to validate if the element exists.

Reply:
@shum should we validate in your opinion?  Thanks to you both.

Reply:
I think GetOk is sufficient as it already checks "whether or not the key has been set to a non-zero value". LGTM.